src/main/java/tc/oc/pgm/rotation/VotingPool.java

```diff
@@ -0,0 +1,91 @@
+package tc.oc.pgm.rotation;
+
+import java.util.*;
+import org.bukkit.configuration.ConfigurationSection;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.map.PGMMap;
+
+@SuppressWarnings("unchecked")
+public class VotingPool extends MapPool {
+
+  // Number of maps in the vote, unless not enough maps in pool
+  private static final int MAX_VOTE_OPTIONS = 5;
```

Comment:
We have to be careful about paradox of choice. What about 3 maps with an additional random map option?

Reply:
The idea is, show 5 maps, all of them at random from the pool, but weighted with their "how liked were they before"

Reply:
I'm not sure I understand

Reply:
Imagine we had just 3 vote options, and a map pool with 4 maps:
 - Map 1: 20%
 - Map 2: 20%
 - Map 3: 20%
 - Map 4: 20%
 - Map 5: 20%

Randomly but based on weight, 3 options are picked (imagine map 1,2,3 are picked).
10 players vote, and give 8, 7, 3 votes to the maps (multiple choice is allowed)
The maps are updated to be:
 - Map 1: ~~90%~~ 0% (since it won the vote, it's picked to play, resets to 0%)
 - Map 2: 70%
 - Map 3: 30%
 - Map 4: 20%
 - Map 5: 20%

Randomly based on weight,3 options are picked (imagine map 3, 4 and 5 are picked)
10 players vote, and give 4 5 1 votes to the maps
 - Map 1: 5% (0 * 0.75 + 20 * 0.25)
 - Map 2: 57.5% (70 * 0.75 + 20 * 0.25)
 - Map 3: 40%
 - Map 4: ~~50%~~ 0% (won the vote, even if map 2 has higher score)
 - Map 5: 10%

With this type of system, there's no non-random picks, all are always polled at random, but the ones who were previously voted are more likely to get picked for voting, but not necessarily what must happen. Also if a map is not voted at all, it will land even lower than the avg, and is less likely to be picked than any other random map that hasn't yet been voted.